### PR TITLE
Fix non existing file links

### LINKED ISSUE #6446: Non existing files throw error 500: the thumbnail configuration could not be found
## Description
Non-existing files trigger an error 500. I can reproduce this in any installation, also on getkirby.com:

https://getkirby.com/media/site/hello/error.pdf
https://getkirby.com/media/pages/features/hello/error.jpeg

Enabling debug modes shows this error message: The thumbnail configuration could not be found.

**Expected behavior**  
A 404 should be returned instead.

## Your setup
**Kirby Version**  
4.2.0

### PR DESCRIPTION
**Pull request by afbora**
### Fixes
- Non existing media files no longer throw error 500: the thumbnail configuration could not be found #6446

### Breaking changes
None

## Ready?

- [x] In-code documentation (wherever needed)
- [x] Unit tests for fixed bug/feature
- [x] Tests and checks all pass

### For review team

- [ ] Add lab and/or sandbox examples (wherever helpful)
- [x] Add changes & docs to release notes draft in Notion

### CODE REVIEW

File: src/Cms/Media.php [hunk 1]
```diff
@@ -57,7 +57,17 @@ public static function link(
 		}
 
 		// try to generate a thumb for the file
-		return static::thumb($model, $hash, $filename);
+		try {
+			return static::thumb($model, $hash, $filename);
+		} catch (Throwable $e) {
```
Comment (lukasbestle) [inline src/Cms/Media.php]: Not sure, but maybe we could restrict this to just the `NotFoundException`. Then all actual errors will still produce HTTP 500 and the fatal page, while the non-existing thumbs will return the error page with HTTP 404. With this change, we also wouldn't need to differentiate between debug and non-debug.

Maybe I'm missing something though, so feel free to correct me :)

Comment (bastianallgeier) [reply src/Cms/Media.php]: I like the idea to really look for the specific NotFoundException

Comment (afbora) [reply src/Cms/Media.php]: @lukasbestle I'm a bit confused. Actually `static::thumb` already throws `NotFoundException` but page status code is 500 🤷‍♂️ 
https://github.com/getkirby/kirby/blob/fix/6446-non-existing-file-links/src/Cms/Media.php#L122-L123

Can you explain what you have in mind, please?

Comment (afbora) [reply src/Cms/Media.php]: Nvm, I get it. I'll implement.

**Comment by afbora** [comment]
I'll squash commits if it's OK.

File: src/Cms/Media.php [hunk 2]
```diff
@@ -57,7 +57,12 @@ public static function link(
 		}
 
 		// try to generate a thumb for the file
-		return static::thumb($model, $hash, $filename);
+		try {
+			return static::thumb($model, $hash, $filename);
+		} catch (NotFoundException) {
+			// render the error page if some stuff for thumb generating not found
```
Comment (lukasbestle) [inline src/Cms/Media.php]:
```suggestion
			// render the error page if there is no job for this filename
```

**Review by lukasbestle — APPROVED** [review]

**Pull request merged**